### DISCUSSION TITLE
How to use REST api using OAuth2 credentials?

### DISCUSSION THREAD

**Discussion by arladmin** [dani-garcia/vaultwarden#3039] [Q&A]
When trying to use a REST API endpoint (e.g. `GET /list/object/collections`), i'm trying to pass on the Oauth2 Client Credentials for authentication.

Base URL: `https://api.my-vault-domain`
Access Token URL: `https://identity.my-vault-domain/connect/token` 

But i keep on getting the error:
```
<html>
<head><title>405 Not Allowed</title></head>
<body>
<center><h1>405 Not Allowed</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

What am i doing wrong here?

**Comment by BlackDex** [comment]
Vaultwarden doesn't support the public API. We only support the client API which is very different.

Also the warning you get is from nginx, not from Vaultwarden.
You probably have done security features enabled like ModSecurity or something. These tend to block some requests like PUT and other stuff.
